src/common/logger.h

```diff
@@ -104,16 +104,14 @@ class Logger
             int minutes = local_tm.tm_min;
             int seconds = local_tm.tm_sec;
 
-            std::stringstream ss;
-            ss << std::setfill('0');
-            ss << year << '-' << std::setw(2) << month << '-' << std::setw(2) << day;
-            ss << ' ';
-            ss << std::setw(2) << hours << ':' << std::setw(2) << minutes << ':' << std::setw(2) << seconds;
-
-            const String timestamp = ss.str().c_str();
+            char buffer[256];
+            #ifndef _WIN32
+            sprintf(buffer, "%04d-%02d-%02d %02d:%02d:%02d", year, month, day, hours, minutes, seconds);
+            #else
+            sprintf_s(buffer, 256, "%04d-%02d-%02d %02d:%02d:%02d", year, month, day, hours, minutes, seconds);
```

Comment:
My C/C++ knowledge is limited (was my first programming language but nowerdays I use it rather seldom as since I work as a software dev I always developed java server applications)
But if buffering was the issue, would it also be possible to use snprintf insetad to save the operating system check?

Like:
```
...
int minutes = local_tm.tm_min;
int seconds = local_tm.tm_sec;

char buffer[256];
snprintf(buffer, sizeof(buffer), "%04d-%02d-%02d %02d:%02d:%02d", year, month, day, hours, minutes, seconds);

String message = buffer;
message += " [" + _get_level(p_level) + "]: ";

...
```

Reply:
Yep, I'll make that adjustment 👍